coding-challenge/tst/bank/bank-account.test.ts

```diff
@@ -86,7 +105,19 @@ describe('Tests for bank account class', () => {
     });
 
     it('should transfer the specified sum back to the source account if the deposit operation fails!', () => {
+        // Act
+        let destinationBankAccount;
+        const transferAmount = 100;
+        
+        // Act & Assert
+        expect(() => {
+            bankAccount.transfer(transferAmount, destinationBankAccount);
+        }).toThrow('Something went wrong! Transaction canceled!');
 
+        // Assert
+        expect(
+            bankAccount.checkBalance()
```

Comment:
We are trying to do unit tests here, if you check the actual implementation of a method from your class you are doing integration tests here, which are out of scope!